src/addrman.cpp

```diff
@@ -1065,6 +1098,17 @@ int AddrManImpl::CheckAddrman() const
     if (nKey.IsNull())
         return -16;
 
+    // It's possible that m_network_counts may have all-zero entries that local_counts
+    // doesn't have if addrs from a network were being added and then removed again in the past.
+    if (m_network_counts.size() < local_counts.size()) {
+        return -20;
+    }
+    for (const auto& [net, count] : m_network_counts) {
+        if (local_counts[net].n_new != count.n_new || local_counts[net].n_tried != count.n_tried) {
```

Comment:
`local_count[net]` could be called on the non-existent key, but that's fine and will return 0, right?

Reply:
perhaps a test for this case would be great...

Reply:
Oh, I should have tried to break it and see whether there is a such test. Explicit is better of course.

Reply:
For this to happen, we need to have an address in AddrMan for a network at one point in time, which later gets removed, so that we have an entry in `m_network_counts` that has only zeroes, but no entry in `local_counts`.
This happens in the `remove_invalid` subtest which tampers with the serialized stream, so that an address gets added and then removed during Deserialization - if you break `CheckAddrman()` by adding an assert at this spot, this subtest will fail.

I think it could also happen theoretically if the last remaining address from network A is being kicked out by an address from network B due to a collision in the new tables. But this doesn't seem trivial to write a unit test for, since addresses from different networks are typically from different groups, so I'm not sure if it's easily possible to write an explicit test for this.